### PR TITLE
Initialize scopes in base policy when loading from cache

### PR DESCRIPTION
**Pull request by christothes**
This fixes an issue where the `ChallengeBasedAuthenticationPolicy` doesn't initialize the base BearerTokenChallengeAuthenticationPolicy Scopes array on requests where the `ChallengeBasedAuthenticationPolicy` is fetching its initial scope from cache.

### CODE REVIEW

File: sdk/keyvault/Azure.Security.KeyVault.Shared/src/ChallengeBasedAuthenticationPolicy.cs [hunk 1]
```diff
@@ -67,6 +67,11 @@ protected async Task PreProcessAsync(HttpMessage message, ReadOnlyMemory<HttpPip
                 // set the content to the original content.
                 message.Request.Content = originalContent;
             }
+            else
+            {
+                // We fetched the scope from the cache, but we have not initialized the Scopes in the base yet.
+                Scopes = _scope.Scopes;
```
Comment (pakrym) [inline sdk/keyvault/Azure.Security.KeyVault.Shared/src/ChallengeBasedAuthenticationPolicy.cs:73]: Test?

File: sdk/keyvault/Azure.Security.KeyVault.Administration/tests/ChallengeBasedAuthenticationPolicyTests.cs [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        public class MockCredentialThrowsWithNoScopes : TokenCredential
+        {
+            public override ValueTask<AccessToken> GetTokenAsync(TokenRequestContext requestContext, CancellationToken cancellationToken)
+            {
+                return new ValueTask<AccessToken>(GetToken(requestContext, cancellationToken));
+            }
+
+            public override AccessToken GetToken(TokenRequestContext requestContext, CancellationToken cancellationToken)
+            {
+                if (requestContext.Scopes.Length != 1)
```
Comment (pakrym) [inline sdk/keyvault/Azure.Security.KeyVault.Administration/tests/ChallengeBasedAuthenticationPolicyTests.cs:51]: `== 0`?

Comment (christothes) [reply sdk/keyvault/Azure.Security.KeyVault.Administration/tests/ChallengeBasedAuthenticationPolicyTests.cs:51]: In practice, we actually throw on GetToken if there is anything but 1 scope.

File: sdk/keyvault/Azure.Security.KeyVault.Administration/tests/ChallengeBasedAuthenticationPolicyTests.cs [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+            var keyvaultChallengeResponse = new MockResponse(401);
+            keyvaultChallengeResponse.AddHeader(new HttpHeader("WWW-Authenticate", KeyVaultChallenge));
+            MockTransport transport = CreateMockTransport(keyvaultChallengeResponse, new MockResponse(200));
+
+            var response = await SendGetRequest(transport, _policy, uri: new Uri("https://example.com"));
+
+            Assert.That(response.Status, Is.EqualTo(200));
+
+            // Construct a new policy so that we can get the Scopes from cache.
+            _policy = new ChallengeBasedAuthenticationPolicy(new MockCredentialThrowsWithNoScopes());
```
Comment (pakrym) [inline sdk/keyvault/Azure.Security.KeyVault.Administration/tests/ChallengeBasedAuthenticationPolicyTests.cs:34]: Should we clean the cache before running this test and make it non-parallelizeable?

**Review by pakrym — APPROVED** [review]

File: sdk/keyvault/Azure.Security.KeyVault.Administration/tests/ChallengeBasedAuthenticationPolicyTests.cs (hunk 3 above)
Comment (christothes) [reply sdk/keyvault/Azure.Security.KeyVault.Administration/tests/ChallengeBasedAuthenticationPolicyTests.cs:34]: Good call.

**Review by heaths — APPROVED** [review]

**Pull request merged**